ckanext/qgov/common/authenticator.py

```diff
@@ -3,27 +3,36 @@
 """
 
 import logging
+from ckan.common import g
 from ckan.lib.authenticator import UsernamePasswordAuthenticator
+from ckan.lib.redis import connect_to_redis
 from ckan.model import User, Session
 
-from sqlalchemy import Column, types, MetaData, DDL
-from sqlalchemy.ext.declarative import declarative_base
-
 from zope.interface import implements
 from repoze.who.interfaces import IAuthenticator
 
-BASE = declarative_base()
-
 LOG = logging.getLogger(__name__)
 
+LOGIN_THROTTLE_EXPIRY = 1800
+
+
+def unlock_account(login_name):
+    """ Unlock an account (erase the failed login attempts).
+    """
+    qgov_user = Session.query(User).filter(User.id == login_name).first()
+    if qgov_user:
+        cache_key = '{}.ckanext.qgov.login_attempts.{}'.format(g.site_id, login_name)
+        redis_conn = connect_to_redis()
+        if redis_conn.get(cache_key):
+            LOG.debug("Clearing failed login attempts for %s", login_name)
+            redis_conn.delete(cache_key)
+    else:
+        LOG.debug("Account %s not found", login_name)
+
 
 def intercept_authenticator():
     """ Replaces the existing authenticate function with our custom one.
     """
-    meta = MetaData(bind=Session.get_bind(), reflect=True)
-    if 'user' in meta.tables and 'login_attempts' not in meta.tables['user'].columns:
-        LOG.warn("'login_attempts' field does not exist, adding...")
-        DDL("ALTER TABLE public.user ADD COLUMN login_attempts SMALLINT DEFAULT 0").execute(Session.get_bind())
```

Comment:
what about cleaning up the extra column post 2.9 upgrade?

Reply:
That can be done manually but it's not hurting anything so is not urgent.